taskcluster/scriptworker_taskgraph/transforms/k8s_image.py

```diff
@@ -0,0 +1,92 @@
+    unexpected discrepancies in upstream deps.
+
+    """
+    for job in jobs:
+        attributes = job["attributes"]
+        dependencies = job.setdefault("dependencies", {})
+        digest_directories = None
+        for dep_task in config.kind_dependencies_tasks:
+            dep_attrs = dep_task.attributes
+            dep_kind = dep_task.kind
+            if dep_attrs["python-version"]  == attributes["python-version"] and \
+                    dep_attrs["script-name"] == attributes["script-name"]:
+                if dependencies.get(dep_kind):
+                    raise Exception("Duplicate kind {kind} dependencies: {existing_label}, {new_label}".format(
+                        kind=dep_kind,
+                        existing_label=dependencies[dep_kind]["label"],
+                        new_label=dep_task.label,
+                    ))
+                dependencies[dep_kind] = dep_task.label
+                if dep_attrs.get("digest-directories"):
+                    if digest_directories and digest_directories != dep_attrs["digest-directories"]:
+                        raise Exception("Conflicting digest_directories: {existing_digest} {new_digest}".format(
+                            existing_digest=digest_directories,
+                            new_digest=dep_attrs["digest-directories"],
+                        ))
+                    digest_directories = dep_attrs["digest-directories"]
+        if digest_directories:
+            attributes["digest-directories"] = digest_directories
+        yield job
+
+
+@transforms.add
+def set_environment(config, jobs):
+    """Set the environment variables for the docker hub task."""
+    for job in jobs:
+        project_name = job["attributes"]["script-name"]
+        secret_url = job.pop("deploy-secret-url")
+        tasks_for = config.params['tasks_for']
+        scopes = job.setdefault("scopes", [])
+        attributes = job["attributes"]
+        env = job["worker"].setdefault("env", {})
+        env["HEAD_REV"] = config.params['head_rev']
+        env["REPO_URL"] = config.params['head_repository']
+        env["PROJECT_NAME"] = project_name
+        env["TASKCLUSTER_ROOT_URL"] = "$TASKCLUSTER_ROOT_URL"
+        env["DOCKER_TAG"] = "unknown"
+        env["DOCKER_REPO"] = job.pop("docker-repo")
+        force_push_docker_image = False
+        if tasks_for == 'github-pull-request':
+            env["DOCKER_TAG"] = "pull-request"
+        elif tasks_for == 'github-push':
+            for ref_name in ("dev", "production"):
+                if config.params['head_ref'] == "refs/heads/{}-{}".format(ref_name, project_name):
+                    env["DOCKER_TAG"] = ref_name
+                    force_push_docker_image = True
+                    break
+            else:
+                if config.params['head_ref'].startswith('refs/heads/'):
+                    env["DOCKER_TAG"] = config.params['head_ref'].replace('refs/heads/', '')
```

Comment:
Is the intention that this catch `refs/heads/{dev,production}`? If so, I'd inclined to be explicit about that. 

Maybe `re.match(r'refs/heads/(dev|production)(?:-[a-z]*)$')` might be closer to what we want. (Thinking about this, maybe it might make sense to make this a graph parameter).

Reply:
Sure. I'm hoping we can follow up with relpro soon, which will remove the need for dev and prod branches. Given that timeframe, I didn't prioritize making these any nicer.

Reply:
I think we do have to set DOCKER_TAG no matter what, or the task will fail. The below check actually checks for whether we want to push to dockerhub.

Reply:
I'm setting `parameters["docker_tag"]` now.